### DISCUSSION TITLE
A bug while grouping code blocks via <pre> with `$ =$` in?

### DISCUSSION THREAD

**Discussion by xihajun** [squidfunk/mkdocs-material#4833] [Q&A] [answered]
When I try to group code blocks using the example below:

**With grouping code blocks**
```
=== "A Bug?"
    <pre><code>cd $(hello) run --repo=$REPO</code></pre> <!--what happens here?-->
```
Output:
![image](https://user-images.githubusercontent.com/25631641/211015877-880677d4-396e-4060-8a01-a5ebdec16f1d.png)

**Without the grouping code blocks things work well**
```
<pre><code>cd $(hello) run --repo=$REPO</code></pre>
```
Output:
![image](https://user-images.githubusercontent.com/25631641/211015979-576d9695-27c7-4228-a60f-a0af8f3def5d.png)

(For some reason we have to use `<pre>` to encode code)

**Comment by xihajun** [comment, marked as answer]
For some reason, `$ -> \( and \)`

**Reply by xihajun** [reply]
Does it enable a math/latex mode?

**Reply by squidfunk** [reply]
Yes, that is very likely. Try disabling the `arithmatex` extensions to see if the error persists.

**Reply by xihajun** [reply]
Thank you very much! Problem solved after removing markdown_extensions `pymdownx.arithmatex`

**Reply by facelessuser** [reply]
Why is disabling `pymdown.arithmatex` necessary?

If you use normal code block syntax, this should work fine:

````
=== "A Bug?"
    ```
    cd $(hello) run --repo=$REPO
    ```
````

Now, why Python Markdown allows you to parse the content of a raw `code` HTML element is curious. I would have thought that Python Markdown would ignore such elements.

Here's what I think the issue is. Python Markdown may not consider this a bug, but more just how things are implemented. Raw HTML parsing, IIRC, is meant to be at the root indentation level of the documentation. Syntax, such as tabs, that require indentation do not have content that meets these criteria.

We can test the following, without tabs, and it works fine:

```html
<pre>
<code>cd $(hello) run --repo=$REPO</code>
</pre>
```

I think this is a flaw with any construct that uses indentation syntax: Lists, Admonitions, Details, Tabs, etc. Or better put, how Python Markdown treats indentation syntax plugins' content.

Personally, I think Python Markdown should treat HTML properly in such plugins, but I think I've complained about this before with not much luck.

Anyway, in this case, fenced blocks are going to be your friend (via `pymdownx.superfences`) opposed to just disabling plugins, unless you don't need Arithmatex, but I imagine other plugins may give you issues as well.
